blockstore/src/main/java/brooklyn/location/blockstore/ec2/Ec2VolumeCustomizers.java

```diff
@@ -69,104 +70,172 @@ public static JcloudsLocationCustomizer withNewVolumes(List<Integer> capacities)
             FilesystemOptions filesystemOptions = new FilesystemOptions("/mnt/brooklyn/"+deviceSuffix);
             volumes.put(blockOptions, filesystemOptions);
         }
-        return withNewVolumes(volumes);
+        return new NewVolumeCustomizer(volumes);
     }
-    
-    public static JcloudsLocationCustomizer withNewVolumes(final Map<BlockDeviceOptions, FilesystemOptions> volumes) {
-        return new BasicJcloudsLocationCustomizer() {
-            @Override
-            public void customize(JcloudsLocation location, ComputeService computeService, TemplateBuilder templateBuilder) {
-                BlockDeviceOptions blockOptions = Iterables.getFirst(volumes.keySet(), null);
-                if (blockOptions != null && blockOptions.getZone() != null) {
-                    templateBuilder.locationId(blockOptions.getZone());
-                }
```

Comment:
The customizer could get serialized, so keep the existing anonymous class - as deprecated and unused, while returning the named class instead (remove return, turning it into a regular expression, then after it do `return new NewVolumeCustomizer(volumes);`.